src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -0,0 +1,270 @@
+import { Cognite3DModel, Cognite3DViewer, THREE } from '@cognite/3d-viewer';
+import * as sdk from '@cognite/sdk';
+import React from 'react';
+import { CacheObject, Callback, MouseScreenPosition } from '../../interfaces';
+import {
+  createViewer as originalCreateViewer,
+  parseBoundingBox,
+  setCameraPosition,
+  ViewerEventTypes,
+} from '../../utils';
+
+let createViewer = originalCreateViewer;
+
+type ClickHandler = (position: MouseScreenPosition) => void;
+
+export interface Model3DViewerProps {
+  projectName: string;
+  modelId: number;
+  revisionId: number;
+  cache: CacheObject;
+  boundingBox: THREE.Box3;
+  assetId?: number;
+  onError?: Callback;
+  onProgress?: Callback;
+  onComplete?: Callback;
+  onReady?: Callback;
+  onClick?: Callback;
+  onCameraChange?: Callback;
+  useDefaultCameraPosition?: boolean;
+}
+
+export function mockCreateViewer(mockFunction: any) {
+  createViewer = mockFunction || originalCreateViewer;
+}
+
+export class Model3DViewer extends React.Component<Model3DViewerProps> {
+  static defaultProps = {
+    boundingBox: parseBoundingBox({
```

Comment:
Will this work? This is an empty box and will always contain zero elements. The initial box should at least be infinitely large.

Reply:
Yes, this is works, I've checked and it was taken from OpInt app